File: backend/src/main/java/com/pizza/delivery/entity/Notification.java
package com.pizza.delivery.entity;

import com.pizza.delivery.enums.NotificationType;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private NotificationType type = NotificationType.SYSTEM;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String message;

    @Column(columnDefinition = "TEXT")
    @Builder.Default
    private String data = "{}";

    @Builder.Default
    private Boolean isRead = false;

    @CreationTimestamp
    private LocalDateTime createdAt;
}

File: android/app/src/main/java/com/pizzadelivery/android/data/model/Notification.kt
package com.pizzadelivery.android.data.model

import com.google.gson.annotations.SerializedName

data class Notification(
    val id: String = "",
    val userId: String = "",
    val title: String = "",
    val message: String = "",
    val type: NotificationType = NotificationType.ORDER_UPDATE,
    val data: Map<String, String>? = null,
    @SerializedName("isRead")
    val isRead: Boolean = false,
    val createdAt: String = ""
)

enum class NotificationType {
    @SerializedName("order_update") ORDER_UPDATE,
    @SerializedName("promotion") PROMOTION,
    @SerializedName("delivery_update") DELIVERY_UPDATE,
    @SerializedName("system") SYSTEM
}
